### DISCUSSION TITLE
8BitDo Retro Mechanical Keyboard and Misc. Keyboards support?

### DISCUSSION THREAD

**Discussion by ABOhiccups** [darthcloud/BlueRetro#837] [Q&A]
Hey DarthCloud,
Do you ever get 8BitDo Retro Mechanical Keyboard (and Misc. Keyboards like iPazzPort Mini Wireless Keyboard) to work with BlueRetro for future firmware update?

**Comment by darthcloud** [comment]
8bitdo KB is supported in latest beta:
https://github.com/darthcloud/BlueRetro/actions/runs/7081676820

It may fix other KB too

**Reply by ABOhiccups** [reply]
Gamepad/Keyboard Presets is not 100% compatible.

I have to go to "BlueRetro Advance config" and in "Output Config" I have to select "Keyboard" under Mode option and save. Now 8BitDo Keyboard is compatible with BlueRetro but with few errors.

Keys = Mapped

` = \\
~ = |
[ = ]
{ = }
| = _

iPazzPort Wireless Mini Keyboard is 99% not compatible.

**Reply by darthcloud** [reply]
What system are you using ?

**Reply by ABOhiccups** [reply]
GameCube

**Reply by darthcloud** [reply]
Which software are you using to test?

Also some of the key you mention are not a main key like |, ~. { .

BlueRetro isn't doing any localisation adaptation, it's just map the keycode of the ASCII GC Keyboard to a standard HID keycode.

I feel what you describe is just what happen when you use a US KB layout with a software that expect a japanese one.

**Reply by darthcloud** [reply]
With the action replay for example, {, }, [, ] are all fine.

**Reply by ABOhiccups** [reply]
I was testing with Phantasy Star Online Episode I & II Plus for GameCube loaded with GC Loader on Swiss.
